xpra/server/auth/exec_auth.py

```diff
@@ -45,12 +49,17 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
     def requires_challenge(self) -> bool:
-        return False
+        return bool(self.http_request)
```

Comment:
Move this to a subclass that overrides the function.
If the `http_request` is missing then authentication should fail.